### ISSUE TITLE
items_after_statements shouldn't fire when const has no name

### ISSUE THREAD

**Issue by SUPERCILEX** [rust-lang/rust-clippy#12304]
### Summary

```
        const _: () = assert!(mem::size_of::<RingKind>() <= u8::BITS as usize);
```

I'm putting this const assertion near the code that uses it to make understanding the assumptions easier. The lint shouldn't fire since there is no name so no way to refer to the const anyway.

### Lint Name

items_after_statements

**Comment by Centri3** [comment]
Makes sense. I assume you'd want to use inline const once that's stabilized, maybe we could add a lint for that as well
